src/BreakpointViewer.h

```diff
@@ -9,51 +9,71 @@
 class QPaintEvent;
 class Breakpoints;
 
+struct BreakpointData {
+	enum Type { BREAKPOINT, WATCHPOINT, CONDITION, ALL } type;
+
+	QString id;
+
+	int row = -1;
+	int index = -1;
+};
+
 class BreakpointViewer : public QTabWidget, private Ui::BreakpointViewer
 {
 	Q_OBJECT
 public:
 	BreakpointViewer(QWidget* parent = nullptr);
 	void setBreakpoints(Breakpoints* bps);
 
-	enum Type { BREAKPOINT, WATCHPOINT, CONDITION };
-
 private:
 	Ui::BreakpointViewer* ui;
 	QTableWidget* selector[3];
+	QMap<QString, BreakpointData> bmap;
+	QMap<QString, BreakpointData> wmap;
+	QMap<QString, BreakpointData> cmap;
+	QMap<QString, BreakpointData>* MapList[3] = {&bmap, &wmap, &cmap};
```

Comment:
The variables `bmap`, `wmap` and `cmap` are nowhere directly accessed(*) (all accesses go via `MapList`).
Wouldn't it be simpler to replace this with (remove the pointer indirection):
```
QMap<QString, BreakpointData> maps[3];
```

Or replace `3` with `BreakPointData::ALL`.  I'm not sure how `ALL` is used in the rest of the code. Maybe consider adding an enum value like `NUM = ALL`.

(*) If one of these 3 maps would be accessed directly, that's still possible via something like `maps[BreakpointData::BREAKPOINT]`.

Reply:
Done!